src/bundler.zig

```diff
@@ -1300,6 +1301,18 @@ pub const Bundler = struct {
                 break :brk logger.Source.initPathString(path.text, "");
             }
 
+            if (strings.startsWith(path.text, "data:")) {
+                const data_url = DataURL.parseWithoutCheck(path.text) catch |err| {
+                    bundler.log.addErrorFmt(null, logger.Loc.Empty, bundler.allocator, "{s} parsing data url \"{s}\"", .{ @errorName(err), path.text }) catch {};
+                    return null;
+                };
+                const body = data_url.decodeData(bun.default_allocator) catch |err| {
```

Comment:
`bun.default_allocator` will leak memory here, instead lets use `this_parse.allocator`
```suggestion
                const body = data_url.decodeData(this_parse.allocator) catch |err| {
```